### PR TITLE
Firebase auth

### PR DESCRIPTION
**Pull request by omjadhav85**
Features added:

1. Added Authentication pages (Login and Signup)
2. Added Firebase authentication calls for login , signup and logout
3. Added Auth Context

### CODE REVIEW

File: src/pages/Authentication/Signup.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    lastName,
+    email,
+    password,
+    confirmPassword,
+    showPassword,
+  } = userDetails;
+
+  const navigate = useNavigate();
+  const { dispatchAuth } = useAuth();
+  // const { dispatchData } = useData();
```
Comment (praveenreddy1798) [inline src/pages/Authentication/Signup.js:31]: remove commented out code if not required

File: src/pages/Dashboard/Dashboard.js [hunk 2]
```diff
@@ -1,58 +1,58 @@
 import { useState } from "react";
-import { NavBar, Footer, ToDoItem, Modal } from "../../components";
+import { ToDoItem, Modal } from "../../components";
 import { useTasks } from "../../contexts/TaskContext";
 import Logo from "../../assets/images/clock_logo.png";
 import { useTitle } from "../../utils/useTitle";
 import { useTheme } from "../../contexts/ThemeContext";
 import { getTheme } from "../../utils/getTheme";
+import { getLocalStorage } from "../../utils/localStorageCalls";
```
Comment (praveenreddy1798) [inline src/pages/Dashboard/Dashboard.js:8]: use barrel imports. Can make your code cleaner

File: src/reducers/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export { setUserDetailsReducer } from "./setUserDetailsReducer";
+export { authReducer } from "./authReducer";
+// export { dataReducer } from "./dataReducer";
```
Comment (praveenreddy1798) [inline src/reducers/index.js:3]: remove the commented out code if not required

**Review by praveenreddy1798 — APPROVED** [review]

File: src/pages/Authentication/Signup.js (hunk 1 above)
Comment (omjadhav85) [reply src/pages/Authentication/Signup.js:31]: I have removed it in my next PR

**Pull request merged**